### PR TITLE
Adds version to otel tracer instrumentation scope

### PR DESCRIPTION
**Pull request by dmarkwat**
To resolve https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/7488#discussion_r1151104185

### CODE REVIEW

**Comment by dmarkwat** [comment]
While I'm here I also added same for opentracing, should that ever get the same javaagent testing treatment as opencensus. OT unfortunately lacks the same metadata class that OC has (`OpenCensusLibraryInformation`) afaict, though, so the value is hard coded. Given the project is effectively inactive, I figured this should be "OK" under the circumstances. However, just say the word and I'll change it or drop it since nothing is affected at this time.

File: opencensus-shim/src/main/java/io/opentelemetry/opencensusshim/OpenTelemetrySpanBuilderImpl.java [hunk 1]
```diff
@@ -49,7 +50,8 @@
 
 class OpenTelemetrySpanBuilderImpl extends SpanBuilder {
   private static final Tracer OTEL_TRACER =
-      GlobalOpenTelemetry.getTracer("io.opentelemetry.opencensusshim");
+      GlobalOpenTelemetry.getTracer(
+          "io.opentelemetry.opencensusshim", OpenCensusLibraryInformation.VERSION);
```
Comment (mateuszrzeszutek) [inline opencensus-shim/src/main/java/io/opentelemetry/opencensusshim/OpenTelemetrySpanBuilderImpl.java:54]: That is the OpenCensus API version, not the version of the OC->OTel shim.

You can read the shim version from the `io/opentelemetry/opencensusshim/version.properties` file - it is autogenerated during build for every library in this project. See https://github.com/open-telemetry/opentelemetry-java/blob/696eea790305eaeba0bbb62a23a43dbbfcf29e32/sdk/common/src/main/java/io/opentelemetry/sdk/resources/Resource.java#L112-L122 for another example

File: opentracing-shim/src/main/java/io/opentelemetry/opentracingshim/TracerShim.java [hunk 2]
```diff
@@ -37,7 +37,9 @@ final class TracerShim implements Tracer {
       TextMapPropagator textMapPropagator,
       TextMapPropagator httpPropagator) {
     this.provider = provider;
-    this.tracer = provider.get("opentracing-shim");
+    // hard code the version for instrumentation scope as no metadata class exists like for
+    // opencensus
+    this.tracer = provider.get("opentracing-shim", "0.33.0");
```
Comment (mateuszrzeszutek) [inline opentracing-shim/src/main/java/io/opentelemetry/opentracingshim/TracerShim.java:42]: Same here, you can read the version from the `version.properties` file

File: opencensus-shim/src/main/java/io/opentelemetry/opencensusshim/OpenTelemetrySpanBuilderImpl.java (hunk 1 above)
Comment (dmarkwat) [reply opencensus-shim/src/main/java/io/opentelemetry/opencensusshim/OpenTelemetrySpanBuilderImpl.java]: Ah geez, what was I thinking? You're right. And I hadn't thought to hunt for the properties file: I'm used to generating `BuildInfo` classes that wrap these so it never to me to look. I've updated these.

**Review by trask — APPROVED** [review]
thx!

**Pull request merged**